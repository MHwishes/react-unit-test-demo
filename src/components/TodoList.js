import {connect} from 'react-redux'
import {toggleTodo} from "../actions";
import PropTypes from "prop-types";
import React, {Component} from "react";
import {visibilityFilters} from '../constant/constants'

export class TodoList extends Component {

    render() {
        const {todos} = this.props;
        return (
            <ul className="todo-list">
                {todos.map(todo =>
                    <li key={todo.id}
                        className={todo.completed ? 'completed' : ''}
                    >
                        <div className="view">
                            <input className={"toggle"+" toggle-"+todo.id} type="checkbox"
                                   disabled={todo.completed}
                                   onClick={() => this.props.toggleTodo({id: todo.id})}/>
                            <label>{todo.text}</label>
                        </div>
                    </li>
                )}
            </ul>
        )
    }
}

TodoList.propTypes = {
    todos: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.number.isRequired,
        completed: PropTypes.bool.isRequired,
        text: PropTypes.string.isRequired
    }).isRequired).isRequired,
};


const getVisibleTodos = (todos, filter) => {
    switch (filter) {
        case visibilityFilters.SHOW_ALL:
            return todos;
        case visibilityFilters.SHOW_COMPLETED:
            return todos.filter(t => t.completed);
        case visibilityFilters.SHOW_ACTIVE:
            return todos.filter(t => !t.completed);
        default:
            throw new Error('Unknown filter: ' + filter)
    }
};

const mapStateToProps = state => ({
    todos: getVisibleTodos(state.todos, state.visibilityFilter)
});

const mapDispatchToProps = {
    toggleTodo
};

export default connect(mapStateToProps, mapDispatchToProps)(TodoList)
