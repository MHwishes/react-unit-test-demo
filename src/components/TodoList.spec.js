import {shallow} from 'enzyme';
import TodoList from './TodoList'
import toJson from "enzyme-to-json";
import configMockStore from 'redux-mock-store';
import React from 'react'

const mockStore = configMockStore();
import {visibilityFilters} from '../constant/constants'

describe('shallow render TodoList component', () => {
    it('render TodoList', () => {
        const initialState = {
            todos: [{id: 1, text: "I am a test Todo", completed: false}],
            visibilityFilter: visibilityFilters.SHOW_ALL
        };
        const store = mockStore(initialState);
        const wrapper = shallow(
            <TodoList store={store}/>);
        expect(toJson(wrapper.dive())).toMatchSnapshot();
    })
});
