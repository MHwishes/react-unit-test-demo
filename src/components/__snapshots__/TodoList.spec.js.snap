// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow render TodoList component render TodoList 1`] = `
<ul
  className="todo-list"
>
  <Connect(Todo)
    completed={false}
    id={1}
    key="1"
    text="I am a test Todo"
  />
</ul>
`;
